### PR TITLE
Feature/webhook dynamic

### PR DESCRIPTION
**Pull request by siddhant3030**
## Summary
 Target issue is #2273 
  - we made webhook headers dynamic 
  - URL also is dynamic. 
  - Added the test case

## Checklist
  Before submitting a pull request, please ensure that you mark these task.

- [x] Ran `mix setup` in the repository root and test.
- [x] If you've fixed a bug or added code that is tested and has test cases. 
- [ ] In the case of adding a new API, you added a **postman** request for that.
- [x] Coding standard and conventions are followed. https://docs.google.com/document/d/1rfU33IjS-ioiIH0TBWpxoLsdyyYdI1tDrqr5HCRIb98/edit?usp=sharing

## Notes
 Please add here if any other information is required for the reviewer.

### CODE REVIEW

**Review by pankaj-ag — CHANGES_REQUESTED** [review]
@siddhant3030 we need to replace the headers and URLs

File: lib/glific/flows/webhook.ex [hunk 1]
```diff
@@ -181,15 +181,40 @@ defmodule Glific.Flows.Webhook do
...
+      results: context.results
+    }
+
+    fields = %{
+      "contact" => Contacts.get_contact_field_map(context.contact_id),
+      "results" => context.results,
+      "flow" => %{name: context.flow.name, id: context.flow.id}
+    }
+
+    MessageVarParser.parse_map(action.headers, fields)
```
Comment (pankaj-ag) [inline lib/glific/flows/webhook.ex]: Can you explain why we need this ?

File: lib/glific/flows/webhook.ex [hunk 2]
```diff
@@ -181,15 +181,40 @@ defmodule Glific.Flows.Webhook do
     nil
   end
 
+  # THis function will create a dynamic headers
+  @spec create_headers(Action.t(), FlowContext.t()) :: map()
+  defp create_headers(action, context) do
+    default_payload = %{
```
Comment (pankaj-ag) [inline lib/glific/flows/webhook.ex]: Let's not duplicate the code and create a common function to get these variables for parser.

**Review by pankaj-ag — CHANGES_REQUESTED** [review]
Please write a test cases also for dynamic header and url

File: lib/glific/flows/webhook.ex (hunk 1 above)
Comment (siddhant3030) [reply lib/glific/flows/webhook.ex]: @pankaj-ag can you point out? Not sure what you are referring to here?

Comment (pankaj-ag) [reply lib/glific/flows/webhook.ex]: Line number 204 - 210 in this file.

image

**Review by pankaj-ag — CHANGES_REQUESTED** [review]

File: lib/glific/flows/webhook.ex [hunk 3]
```diff
@@ -181,20 +181,33 @@ defmodule Glific.Flows.Webhook do
...
+    %{header: header, url: url}
+  end
+
   @spec do_oban(Action.t(), FlowContext.t(), tuple()) :: any
   defp do_oban(action, context, {map, body}) do
-    headers =
-      if is_nil(action.headers),
-        do: %{},
-        else: action.headers
+    dynamic_headers = create_headers(action, context)
```
Comment (pankaj-ag) [inline lib/glific/flows/webhook.ex]:
```suggestion
   parsed_attrs = parse_header_and_url(action, context)
```

**Review by pankaj-ag — CHANGES_REQUESTED** [review]

File: lib/glific/flows/webhook.ex (hunk 3 above)
Comment (siddhant3030) [reply lib/glific/flows/webhook.ex]: Done

**Review by pankaj-ag — APPROVED** [review]

**Pull request merged**